tests/test_cli.py

```diff
@@ -154,21 +156,54 @@ def test_building_from_json_source_provider(self, is_registry_running, temp_imag
         assert excinfo.value.code == 0
         dt.remove_image(temp_image, noprune=True)
 
-    def test_create_build_image(self, temp_image_name):
+    @pytest.mark.parametrize('build_json', [
+        {'valid_json': False, 'valid_schema': False, 'contents':
+            'this is not a json'},
+        {'valid_json': True, 'valid_schema': False, 'contents':
+            '{ "this": "is", "json": "but", "its": "invalid" }'},
+        {'valid_json': True, 'valid_schema': True, 'contents':
+            '{"image": "image_name", "source": {"provider": "path", "uri": "bar"}}'},
+    ])
+    def test_json_validation(self, is_registry_running, temp_image_name, build_json):
         if MOCK:
             mock_docker()
 
+        fd, temp_json_file_path = tempfile.mkstemp()
```

Comment:
How about `from tempfile import NamedTemporaryFile` and `with NamedTemporaryFile(...) as ...:`?